### ISSUE TITLE
False positive PropertyNotSetInConstructor with templates involved

### ISSUE THREAD

**Issue by gregor-tb** [vimeo/psalm#10097]
see: https://psalm.dev/r/bcca3d8b07

This does not report when:
* removing `@var A<int>`
* change `private $member;` to `protected $member;`

But both "solutions" are bad, protected in final class is reported on other rules/tools, and removing @var triggers MissingPropertyType warning.

This issue seems similar to various other _PropertyNotSetInConstructor_ reports, but in this case there are no traits used.

**Comment by stof** [comment]
the message says `Property B::$member is not defined in constructor of B or in any private or final methods called in the constructor`

But here, the call in the constructor happens in the parent class, which is probably why it is not detected.

**Comment by weirdan** [comment]
> But here, the call in the constructor happens in the parent class, which is probably why it is not detected.

No, it works as expected with scalars (https://psalm.dev/r/05f7b6687c) and even named classes, but fails with generics.
